src/core/ngx_connection.c

```diff
@@ -487,8 +487,17 @@ ngx_open_listening_sockets(ngx_cycle_t *cycle)
                 continue;
             }
 
-            s = ngx_socket(ls[i].sockaddr->sa_family, ls[i].type, 0);
+            s = (ngx_socket_t) -1;
+            if (ls[i].protocol > 0) {
+                s = ngx_socket(ls[i].sockaddr->sa_family, ls[i].type,
+                               ls[i].protocol);
+            }
 
+            /* In case of error, retry with the default protocol */
```

Comment:
I think it is best to move it in the block above, after having called `ngx_socket`, no?
It is only in this case that we will retry in case of error.

(or we log the error (not critical) and mention we will retry with the default protocol?)